core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -121,8 +125,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
           context.reply(RegisterExecutorFailed("Duplicate executor ID: " + executorId))
         } else {
           logInfo("Registered executor: " + executorRef + " with ID " + executorId)
-          context.reply(RegisteredExecutor)
-
+          context.reply(RegisteredExecutor(latestTokens))
```

Comment:
The only place where I see tokens being sent to YARN is before the AM is launched (in Client.scala, `setupSecurityToken`). After that it seems all containers are launched with those tokens, so Hari's code would be necessary.

Unless there's a way to update those tokens?